File: web/src/domain/Movies.ts
export interface Movies {
    id: number;
    title: string;
    year: number;
    genre: string[];
    budget: number;
    popularity: number;
    posterPath: string;
    productionCompanies: ProductionCompany[];
    releaseDate: string;
    revenue: number;
    runtime: number;
    status: string;
    tagLine: string;
    productionCountries: string[];
    overview: string;
}

export interface ProductionCompany {
    id: number;
    logo: string;
    name: string;
    originCountry: string;
    movieId: string;
}

export interface MovieImages {
    id: number;
    backdrops: ImageDetail[];
    posters: ImageDetail[];
}

export interface ImageDetail {
    aspect_ratio: number;
    filePath: string;
    height: number;
    iso_639_1: string | null;
    voteAverage: number;
    voteCount: number;
    width: number;
}

export interface TrendingMovies {
    id: number;
    title: string;
    year: number;
    posterPath: string

}

File: app/models/movies.py
from datetime import datetime
from typing import List, Optional
from pydantic import BaseModel

from app.models.production_companies import ProductionCompany


class Movies(BaseModel):
    id: int
    title: str
    year: int
    genre: List[str]
    budget: Optional[float] = None
    popularity: Optional[float] = None
    poster_path: Optional[str] = None
    production_companies: Optional[List[ProductionCompany]] = None
    release_date: datetime
    revenue: int
    runtime: int
    tagline: str
    production_countries: List[str]
    overview: str
    
    
    
    